### DISCUSSION TITLE
Quarto with new input languages other than Markdown only

### DISCUSSION THREAD

**Discussion by aksigkvgithub** [quarto-dev/quarto-cli#6386] [Feature Requests]
### Description

Dear Quarto,

Are you not almost a repetition of Rmarkdown. You might say that: no, we can do Rmarkdown along with other source languages too, and output in different formats of various source languages.

I had written a book in Rmarkdown 3 years back, and got it published with an Indian publisher. Now, an international publisher has given me an offer to publish it's revised  version, provided that the book be in Docbook format. 

I revised the manuscript in Rmarkdown itself and opened my revised Rmarkdown files in Quarto and converted them with Docbook format as well as with Asciidoc and Asciidocdor formats. The latter formats were again converted to Docbook using Asciidocdor and sent to the publisher.

I am very sorry to say that when I looked at bad remarks in the publisher's reply, I had to withdraw my manuscript, because all the formats were not converted correctly and they were really in very bad shapes.

I couldn't understand the Docbook formats, so I got them checked by some experts. But, Asciidoc and Asciidocdor formats I checked myself because I am now studying Asciidocdor for writing new book.They were really bad.

After reflecting within myself,  I came to realise that when Quarto cannot read a format, except Markdown, how can it convert to a format correctly.

Therefore, I request Quarto to develop reading formats of important languages too like that of Asciidoc / Asciidocdor to start with apart from others, so that correct Docbook may be created from Quarto too, not to mention that Docbook are very much demanded by many publishers.

Otherwise, development of Quarto seems to be mere repetition of Rmarkdown. I would suggest that develop Rmarkdown for R users, while make Quarto a versatile document processor for all source languages with all popular markup languages equipped with both input as well as output formats.

With regards,

Dr. A. K. Singh

**Comment by dragonstyle** [comment]
I'm really sorry to hear that you had this experience with our Asciidoc output. Without knowing more about your project, it is hard for me to explain the issues that you ran into, though I think the conclusion `I came to realise that when Quarto cannot read a format, except Markdown, how can it convert to a format correctly.` doesn't really hold up to scrutiny since Quarto starts with markdown and produces many output formats like HTML, LaTeX etc, even though those are not valid input formats.

I think one possible issue that you ran into - prior to Quarto 1.3, asciidoc books were not well supported, but in Quarto 1.3 we added support for the asciidoc/asciidoctor output format for books. At least one book was submitted to a publisher using that output format of a Quarto book and they were pleased with the quality of the generated asciidoc. Perhaps the problem is as simple as using an older version of Quarto which did not support asciidoc for book projects. 

If you'd like I'm happy to help troubleshoot the issues if you can provide a repo or example documents which produce the malformed asciidoc output.

**Comment by aksigkvgithub** [comment]
Thanks for your offer of support. But, your help will not help me because there are too many problems. I am now in the process of manually converting my book to Asciidocdor format.

The only way you can help me for other works is by introducing new feature of including Asciidoc (Asciidocdor) as input language in Quarto so that it can produce pdf through latex route. Asciidocdor produces very good html and Docbook. However , they don't produce good pdf and they don't have reliable LaTeX backend.. Thus , all solution would come in one feature only, if you listen to my request. We shall be able to write in Asciidoc format using Quarto, and would be able to produce a nice pdf through LaTeX backend of Quarto in which both Rmarkdown and Quarto are very good, and through Asciidocdor or Quarto both we can produce html. Docbook can be produced through Asciidocdor. Only one feature would make Quarto most versatile editor.

Thanks and regards,

Dr. A. K. Singh

**Reply by dragonstyle** [reply]
I appreciate you taking the time to reply. Did you ever attempt to confirm the Quarto version that caused you such problem with the asciidoc output?

It isn't currently on our roadmap to support `asciidoc` as an input format (and I have to say I still don't see how you find this related to the quality of our output - you seem happy with our LaTeX output despite the fact that you aren't using LaTeX as an input language). The whole concept of Quarto is take Quarto flavored markdown and emit a variety of output formats (_not_ to take a variety of inputs formats) so I don't see how this would make sense for us.

You could certainly try using Pandoc directly, as it is much more focused on this design center.

Best of luck...

**Comment by aksigkvgithub** [comment]
That's where , I think, Quarto is wasting it's costly time . All these efforts could have been for enhancing Rmarkdown. When Quarto is different application, why to aim at similar objectives,  and not for broader objectives.

You might not agree that Markdown has various loopholes, which it tries to somehow fill up with twisted syntax. It has so far not been able to put a continuation syntax for a numbered list, to cite only one example, whereas there are many.

You might say, no Quarto has @ syntax for the same. I was happy at that time when I saw this in Quarto. But, when I tried it in an article, there was a full-page-table at a particular list-number of the article. In pdf output as well as in html both, the numbered list broke, the list again started at number 1. I used @ in the list before the table and after the table. Still the list broke. By the way I was using Quarto 1.4, the release candidate..

This is a problem of markdown. For such needs it is customary to apply latex or html to resolve i.e. again a kind of twisted syntax approach. I have written a book in Rmarkdown with such twisted approaches only

What I think, Markdown is for very simple articles, And for simple books, and not for those belonging to more complex ones like those belonging to statistics and mathematics.

That's why I was requesting for such feature of input languages as Asciidocdor, for such needs apart from Markdown.

You have such a nice team and more particularly devoted to statistical cause. You have such a brilliant backend of LaTeX. Then, why not take an easy input language for complex job like Asciidocdor. They have good input language, good html and good Docbook. But, they don't have good pdf or LaTeX backends. Pandoc Docbook to pdf also doesn't help very much. And they are not devoted to Statistical cause too.

That's why I was requesting this feature from Quarto, not even from Rmarkdown because latter's name itself includes Markdown.

Thanks and regards,

Dr. A. K. Singh.
